components/date-time/time.js

```diff
@@ -8,87 +8,158 @@ import moment from 'moment';
+			this.syncState( this.props );
 			return;
 		}
-		const newDate = selected.clone().minutes( value );
-		onChange( newDate );
-	};
+		const newDate = date.clone().minutes( value );
+		this.setState( { date: newDate } );
+		const formattedDate = newDate.format( TIMEZONELESS_FORMAT );
+		onChange( formattedDate );
+	}
 
-	const setAM = () => {
-		if ( am === 'AM' ) {
-			return;
-		}
-		const newDate = selected.clone().hours( parseInt( hours, 10 ) % 12 );
-		onChange( newDate );
-	};
+	updateAmPm( value ) {
+		return () => {
+			const { onChange } = this.props;
+			const { am, date, hours } = this.state;
+			if ( am === value ) {
+				return;
+			}
+			let newDate;
+			if ( value === 'PM' ) {
+				newDate = date.clone().hours( ( ( parseInt( hours, 10 ) % 12 ) + 12 ) % 24 );
+			} else {
+				newDate = date.clone().hours( parseInt( hours, 10 ) % 12 );
+			}
+			this.setState( { date: newDate } );
+			const formattedDate = newDate.format( TIMEZONELESS_FORMAT );
+			onChange( formattedDate );
+		};
+	}
 
-	const setPM = () => {
-		if ( am === 'PM' ) {
-			return;
-		}
-		const newDate = selected.clone().hours( ( ( parseInt( hours, 10 ) % 12 ) + 12 ) % 24 );
-		onChange( newDate );
-	};
-
-	return (
-		<div className="components-time-picker">
-			<input
-				className="components-time-picker__input"
-				type="text"
-				value={ hours }
-				onChange={ updateHours }
-			/>
-			<span className="components-time-picker__separator">:</span>
-			<input
-				className="components-time-picker__input"
-				type="text"
-				value={ minutes }
-				onChange={ updateMinutes }
-			/>
-			{ is12Hour && <div>
-				<Button
-					className="button components-time-picker__am-button"
-					isToggled={ am === 'AM' }
-					onClick={ setAM }
-				>
-					{ __( 'AM' ) }
-				</Button>
-				<Button
-					className="button components-time-picker__pm-button"
-					isToggled={ am === 'PM' }
-					onClick={ setPM }
-				>
-					{ __( 'PM' ) }
-				</Button>
-			</div> }
-		</div>
-	);
+	onChangeHours( event ) {
+		this.setState( { hours: event.target.value } );
+	}
+
+	onChangeMinutes( event ) {
+		this.setState( { minutes: event.target.value } );
+	}
+
+	render() {
+		const { is12Hour } = this.props;
+		const { minutes, hours, am } = this.state;
+
+		return (
+			<div className="components-time-picker">
+				<input
+					className="components-time-picker__input"
+					type="text"
+					value={ hours }
+					onChange={ this.onChangeHours }
+					onBlur={ this.updateHours }
+				/>
+				<span className="components-time-picker__separator">:</span>
+				<input
+					className="components-time-picker__input"
+					type="text"
+					value={ minutes }
+					onChange={ this.onChangeMinutes }
+					onBlur={ this.updateMinutes }
+				/>
+				{ is12Hour && <div>
+					<Button
+						className="button components-time-picker__am-button"
+						isToggled={ am === 'AM' }
+						onClick={ this.updateAmPm( 'AM' ) }
+					>
+						{ __( 'AM' ) }
```

Comment:
Not sure if these need some translation context in general. cc @yoavf

Reply:
>Not sure if these need some translation context in general.

I would have possibly expected context to be good here, but noting elsewhere in core it is not provided:

https://github.com/WordPress/wordpress-develop/blob/f3d3254/src/wp-includes/class-wp-locale.php#L197-L201